actors/miner/tests/move_partitions_test.rs

```diff
@@ -0,0 +1,325 @@
+use fil_actor_miner::{
+    deadline_available_for_compaction, deadline_available_for_optimistic_post_dispute,
+    deadline_is_mutable, new_deadline_info, DeadlineInfo,
+};
+use fil_actors_runtime::{
+    runtime::{DomainSeparationTag, RuntimePolicy},
+    test_utils::{expect_abort_contains_message, MockRuntime},
+};
+use fvm_ipld_bitfield::BitField;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::randomness::Randomness;
+use fvm_shared::{clock::ChainEpoch, error::ExitCode};
+
+mod util;
+use util::*;
+const PERIOD_OFFSET: ChainEpoch = 100;
+
+fn setup() -> (ActorHarness, MockRuntime) {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let rt = h.new_runtime();
+    h.construct_and_verify(&rt);
+    rt.balance.replace(BIG_BALANCE.clone());
+
+    (h, rt)
+}
+
+// returns the nearest epoch such that synchronous post verification is required
+fn nearest_unsafe_epoch(rt: &MockRuntime, h: &ActorHarness, from_deadline_id: u64) -> i64 {
+    let current_ddl = h.current_deadline(rt);
+
+    for i in *rt.epoch.borrow().. {
+        if !deadline_available_for_compaction(
+            &rt.policy,
+            current_ddl.period_start,
+            from_deadline_id,
+            i,
+        ) && deadline_available_for_optimistic_post_dispute(
+            &rt.policy,
+            current_ddl.period_start,
+            from_deadline_id,
+            i,
+        ) {
+            return i;
+        }
+    }
+
+    panic!("impossible path");
+}
+
+// returns the nearest epoch such that no synchronous post verification is necessary
+fn nearest_safe_epoch(rt: &MockRuntime, h: &ActorHarness, from_deadline_id: u64) -> i64 {
+    let current_ddl = h.current_deadline(rt);
+
+    for i in *rt.epoch.borrow().. {
+        if deadline_available_for_compaction(
+            &rt.policy,
+            current_ddl.period_start,
+            from_deadline_id,
+            i,
+        ) {
+            return i;
+        }
+    }
+
+    panic!("impossible path");
+}
+
+// returns the farthest deadline from current that satisfies deadline_available_for_move
```

Comment:
Yeah because as you can see, this function only considers the relationship between `current_deadline` and `from_deadline_id`(or `orig_deadline_id`), not `to_deadline_id`(or `dest_deadline_id`).